src/main/thrift/parquet.thrift

```diff
@@ -373,6 +453,51 @@ struct JsonType {
 struct BsonType {
 }
 
+/**
+ * Physical type and encoding for the geometry type.
+ */
+enum GeometryEncoding {
+  /**
+   * Allowed for physical type: BYTE_ARRAY.
+   *
+   * Well-known binary (WKB) representations of geometries. It supports 2D or
+   * 3D geometries of the standard geometry types (Point, LineString, Polygon,
+   * MultiPoint, MultiLineString, MultiPolygon, and GeometryCollection). This
+   * is the preferred option for maximum portability.
+   *
+   * This encoding enables GeometryStatistics to be set in the column chunk
+   * and page index.
+   */
+  WKB = 0;
+
+  // TODO: add native encoding from GeoParquet/GeoArrow
+}
+
+/**
+ * Geometry logical type annotation (added in 2.11.0)
+ */
+struct GeometryType {
+  /**
+   * Physical type and encoding for the geometry type. Please refer to the
+   * definition of GeometryEncoding for more detail.
+   */
+  1: required GeometryEncoding encoding;
+  /**
+   * Edges of polygon.
+   */
+  2: required Edges edges;
+  /**
+   * Coordinate Reference System, i.e. mapping of how coordinates refer to
+   * precise locations on earth, e.g. OGC:CRS84
+   */
+  3: optional string crs;
```

Comment:
@paleolimbot @wgtmac @Kontinuation @zhangfengcdt Please see the latest comment from @desruisseaux on the Iceberg Geometry PR (https://github.com/apache/iceberg/issues/10260). I also discussed this issue with the GeoParquet community today.

In short, some people like projjson given its popularity and completeness, some people don't like it because it is still incomplete and there is another on-going effort at OGC CRS working group called CRSJSON to replace it, some other people like SRID (comments from Snowflake).

We'd better define what CRS encoding is used in this optional CRS string by using an additional enum field to avoid future confusion.

Example:

```
5. optional enum CRSEncoding {
  WKT2 = 0;
  SRID = 1; // Format: AUTHORITY:CODE
  PROJJSON = 2;
  // Future work when CRSJSON is completed: add CRSJSON = 3
}
```

Some members from the GeoParquet community think allowing multiple encoding of CRS introduce additional overhead to the implementer. I would argue that this is not a problem because:
1. The reader / writer does not need to support all CRS Encodings. It is optional anyway.
2. Even if we use projjson, a large number of engines in the Java world  (Hive, Trino, Presto, Sedona/Spark, Sedona/Flink) cannot comprehend it because no Java library for it.
3. The CRS encoding is purely a description field. If an engine does not understand it, just carry it along the way.

In addition, the next GeoParquet community sync is August 12. Please join us for the discussion of this topic: Zoom meeting: https://github.com/opengeospatial/geoparquet/discussions/240

Reply:
+1 for using string type if the encoding is just informative.

Reply:
Looks like strong opinions on all fronts.  +1 for adding the crs type and for it to be a string, in my thought the storage should not be too opinionated.  We can document common values in the docs maybe?

Reply:
@wgtmac According to the feedback from the Iceberg thread, I suggest we also add a string field namely `crs_kind` in addition to the `crs` field. The only allowed value currently is `projjson`. In the future, if there is a new OGC standard called `CRSJSON` that differs from `projjson`, we will allow another value `crsjson`.

For WKT2 2019 <-> PROJJSON, we will implement a Java version of this library https://github.com/rouault/projjson_to_wkt so whoever wants to use WKT2 2019 CRS can use it to get it from the projjson string.

Reply:
Isn't a bit weird that the only allowed value is the non-standard encoding (admittedly depending on what we call "standard"), and to disallow the international "de jure" standard?

> If this is included as an option it would need to be more explicit about what kind of WKT2 we're talking about

This is not needed as latest WKT 2 is backward compatible with the previous version. The ambiguity was on the different kinds of WKT 1.

Reply:
@szehon-ho According to GeoParquet spec (https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md), when I say `default to OGC:CRS84`, the `crs` field actually stores this JSON value:

```
{
    "$schema": "https://proj.org/schemas/v0.5/projjson.schema.json",
    "type": "GeographicCRS",
    "name": "WGS 84 longitude-latitude",
    "datum": {
        "type": "GeodeticReferenceFrame",
        "name": "World Geodetic System 1984",
        "ellipsoid": {
            "name": "WGS 84",
            "semi_major_axis": 6378137,
            "inverse_flattening": 298.257223563
        }
    },
    "coordinate_system": {
        "subtype": "ellipsoidal",
        "axis": [
        {
            "name": "Geodetic longitude",
            "abbreviation": "Lon",
            "direction": "east",
            "unit": "degree"
        },
        {
            "name": "Geodetic latitude",
            "abbreviation": "Lat",
            "direction": "north",
            "unit": "degree"
        }
        ]
    },
    "id": {
        "authority": "OGC",
        "code": "CRS84"
    }
}
```

And the `crs_kind` field is `PROJJSON`.

Both of `crs` and `crs_kind` field are optional but they both need to present together, if one needs to store CRS info

Reply:
> what do we put for crs_kind when crs="OGC:CRS84". It is empty ?

+1 to all @jiayuasu said, and just to be totally clear - it would often be empty. Including the CRS and kind/encoding in this case is more 'informative' - implementations should understand that if they see crs="OGC:CRS84" then they don't need to check the crs and kind/encoding, and if the values differ then they should use CRS84 and ignore the provided CRS. We should provide the definition of OGC:CRS84 in all possible encodings in a link from the core definition - WKT1, WKT2, PROJJSON, etc. so that any projection aware library is sure to get the exact right definition. The goal for GeoParquet was to make it so that implementations that only want to support long / lat can without having to parse / worry about anything else, since it's a lot of complexity and requires some sort of geo library to parse.

Reply:
I have added the above example to the `crs` and `crs_encoding` fields. Please check @jiayuasu @cholmes